### DISCUSSION TITLE
Tool to measure memory usage inside the enclave.

### DISCUSSION THREAD

**Discussion by victormassy** [gramineproject/gramine#1086] [Q&A]
Hey, 

I'm looking for a tool to measure the memory used by my application inside the enclave. Do you know anything that could do the job? 

Thank you

**Comment by mkow** [comment]
You can print `/proc/<pid>/status` and check `VmPeak:` field, we emulate it inside Gramine.

**Reply by mkow** [reply]
(but note that it's not the _current_ usage, it's the _peak_ usage since the in-Gramine process started)

**Reply by victormassy** [reply]
Do you mean checking the status of the process that calls gramine-sgx ? How can I find the process ID ?

**Reply by victormassy** [reply]
I've tried something like that  `gramine-sgx main && cat /proc/$$/status` , but I'm not sure if this really check the process inside the enclave.

**Reply by mkow** [reply]
No, I mean reading this file from _inside_ of Gramine.

**Reply by victormassy** [reply]
Thank you for your answer ! I've conducted some tests. VmPeak for my application is always around 1GB no matter the size of the input file. It seems that the memory allocated to one specific process is decided before execution. Is it true?

**Reply by mkow** [reply]
I don't know what your application does nor how it's configured, so I can't tell ¯\\\_(ツ)\_\/¯ It's your application allocating the memory.
